File: ios/RoutineChart/RoutineChart/Core/Domain/Models/ChildProfile.swift
//
//  ChildProfile.swift
//  RoutineChart
//
//  Domain model for ChildProfile entity
//

import Foundation

struct ChildProfile: Identifiable, Codable, Equatable, Hashable {
    let id: String
    let familyId: String
    var displayName: String
    var avatarIcon: String?
    var ageBand: AgeBand
    var readingMode: ReadingMode
    var audioEnabled: Bool
    let createdAt: Date
    
    init(
        id: String = UUID().uuidString,
        familyId: String,
        displayName: String,
        avatarIcon: String? = nil,
        ageBand: AgeBand,
        readingMode: ReadingMode,
        audioEnabled: Bool = true,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.familyId = familyId
        self.displayName = displayName
        self.avatarIcon = avatarIcon
        self.ageBand = ageBand
        self.readingMode = readingMode
        self.audioEnabled = audioEnabled
        self.createdAt = createdAt
    }
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/models/ChildProfile.kt
package com.HammersTech.RoutineChart.core.domain.models

import java.time.Instant

/**
 * ChildProfile domain model
 * Represents a child's profile with preferences for routine presentation
 */
data class ChildProfile(
    val id: String,
    val familyId: String,
    val displayName: String,
    val avatarIcon: String?,
    val ageBand: AgeBand,
    val readingMode: ReadingMode,
    val audioEnabled: Boolean,
    val createdAt: Instant,
) {
    companion object {
        const val DEFAULT_AUDIO_ENABLED = true
    }
}
